### ISSUE TITLE
Add support for GSON 2.11.0

### ISSUE THREAD

**Issue by Donnerbart** [javers/javers#1377]
# Clear description of my expectations versus reality

We have a project that uses these dependencies:
* `com.google.code.gson:gson:2.10.1`
* `org.javers:javers-core:5.15.0`

With our latest dependency update we bump GSON to `2.11.0` and start seeing this exception in our application:
```java
Cannot override built-in adapter for class com.google.gson.JsonElement
java.lang.IllegalArgumentException: Cannot override built-in adapter for class com.google.gson.JsonElement
	at com.google.gson.GsonBuilder.registerTypeAdapter(GsonBuilder.java:714)
	at org.javers.core.json.JsonConverterBuilder.registerNativeGsonSerializer(JsonConverterBuilder.java:79)
	at org.javers.core.json.JsonConverterBuilder.registerJsonTypeAdapterForType(JsonConverterBuilder.java:169)
	at org.javers.core.json.JsonConverterBuilder.lambda$registerJsonTypeAdapter$0(JsonConverterBuilder.java:127)
	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4305)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1116)
	at org.javers.core.json.JsonConverterBuilder.registerJsonTypeAdapter(JsonConverterBuilder.java:127)
	at org.javers.core.json.JsonConverterBuilder.registerJsonTypeAdapters(JsonConverterBuilder.java:115)
	at org.javers.core.JaversBuilder.bootJsonConverter(JaversBuilder.java:804)
	at org.javers.core.JaversBuilder.assembleJaversInstance(JaversBuilder.java:154)
	at org.javers.core.JaversBuilder.build(JaversBuilder.java:126)
```
Using this builder:
```java
    private final @NotNull Javers javers = JaversBuilder.javers()
            .registerJaversRepository(new InMemoryRepository())
            .withListCompareAlgorithm(LEVENSHTEIN_DISTANCE)
            .build();
```

The expectation is that Javers works with Gson 2.11.0.
The reality is that Javers stops working with Gson 2.11.0.

# Steps To Reproduce

I have a **runnable test case** which isolates the bug and allows Javers Core Team to easily reproduce it. I have pushed this test case to my fork of this repository:

https://github.com/Donnerbart/javers/blob/master/javers-core/src/test/groovy/org/javers/core/JaversBuilderTest.groovy

(every existing test should fail after [bumping Gson to 2.11.0](https://github.com/javers/javers/compare/master...Donnerbart:javers:master))

# Javers' Version

5.15.0

**Comment by Donnerbart** [comment]
Also tested with Javers 7.4.2, breaks with the same exception `java.lang.IllegalArgumentException: Cannot override built-in adapter for class com.google.gson.JsonElement`.

**Comment by bartoszwalacik** [comment]
Right, seems like like Gson 2.11.0 introduced breaking changes  (vs 2.10) which break Javers

**Comment by bartoszwalacik** [comment]
I will fix it, thanks for reporting @Donnerbart

**Comment by bartoszwalacik** [comment]
https://github.com/javers/javers/pull/1379

**Comment by lakkepogusyam** [comment]
@bartoszwalacik  can you check the Post No 1380  can you solve it

**Comment by bartoszwalacik** [comment]
the fix is released in 7.4.3
